packages/cozy-harvest-lib/src/models/ConnectionFlow.spec.js

```diff
@@ -369,138 +369,67 @@ describe('ConnectionFlow', () => {
       )
     })
 
-    it('should call ensureTriggerAndLaunch with account', async () => {
+    it('should call ensureTrigger with account', async () => {
       const { flow, client } = setup()
-      jest
-        .spyOn(flow, 'ensureTriggerAndLaunch')
-        .mockResolvedValue(fixtures.launchedJob)
 
       await setupSubmit(flow, {
         account: fixtures.existingAccount,
         trigger: fixtures.existingTrigger,
         userCredentials: fixtures.credentials
       })
 
-      expect(flow.ensureTriggerAndLaunch).toHaveBeenCalledWith(
+      expect(ensureTrigger).toHaveBeenCalledWith(
         client,
         expect.objectContaining({ account: fixtures.updatedAccount })
       )
     })
```

Comment:
Since now this `ensureTiggerAndLaunch` has been split into several call, you should also test that `ensureDefaultFolderPathInAccount` has been called for instance. 

Once it's done, I'm interested in what is the expected behavior if `ensureTrigger` resole, but `ensureDefaultFolderPathInAccount` throw.

Reply:
Done.

Concerning `ensureDefaultFolderPathInAccount`, it depends on saveAccount, which retries on 409 errors and it does not throw (I fixed the saveAccount outside the try catch), errors are converted to warning messages. 

In this case (no default folder path defined), it can be defined once again on next account  update or the stack has a mecanism of defining a default folder with the name as the slug of the connector.